src/lib/resource.js

```diff
@@ -1,51 +1,139 @@
+  if (Object.keys(updateObj).length > 0) {
+    const updatedCnt = await Resource.update(updateObj, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });
+
+    // If we don't update anything throw an error to retry.
+    if (updatedCnt[0] === 0) {
+      throw Error('Failed to update resource.');
+    }
+
+    return updatedCnt[0];
+  }
+
+  return 0;
+};
+
+const getMetadataValues = async (resourceId, resourceUrl, isEclkc) => {
+  // Attempt to get the resource metadata (if valid ECLKC resource).
+  // Sample: https://eclkc.ohs.acf.hhs.gov/mental-health/article/head-start-heals-campaign?_format=json
+  if (isEclkc) {
+    try {
+      const metadataUrl = `${resourceUrl}?_format=json`;
+      const res = await axios.get(metadataUrl, { maxRedirects: 3 });
+      const metadata = res.data;
+
+      // get created.
+      const { created } = metadata;
+      // get changed.
+      const { changed } = metadata;
+      // get title.
+      const { title } = metadata;
+      // get field_taxonomy_national_centers.
+      // eslint-disable-next-line max-len
+      const fieldTaxonomyNationalCenters = metadata.field_taxonomy_national_centers;
+      // get the field_taxonomy_topic.
+      const fieldTaxonomyTopic = metadata.field_taxonomy_topic;
+      // get the langcode.
+      const { langcode } = metadata;
+      // get the field_context.
+      const fieldContext = metadata.field_context;
+
+      // Create the metadata object.
+      const metadataObj = {
+        created,
+        changed,
+        title,
+        fieldTaxonomyNationalCenters,
+        fieldTaxonomyTopic,
+        langcode,
+        fieldContext,
+      };
+
+      // Update URL in DB.
+      await Resource.update({
+        metadata: metadataObj,
+        metadataUpdatedAt: new Date(),
+      }, {
+        where: { url: resourceUrl },
+        individualHooks: false,
+      });
+    } catch (error) {
+      auditLogger.error(`Resource Queue: Unable to retrieve ECLKC metadata for Resource (ID: ${resourceId} URL: ${resourceUrl}).`, error);
+    }
+  } else {
+    auditLogger.info(`Resource Queue: Warning, not a ECLKC resource SKIPPING metadata collection for: '${resourceUrl}'.`);
+  }
+};
+
+const getResourceMetaDataJob = async (job) => {
+  const {
+    resourceId, resourceUrl,
+  } = job.data;
+
+  try {
+    // 1.) Determine if this is an ECLKC resource.
+    const isEclkc = resourceUrl.includes('eclkc.ohs.acf.hhs.gov');
+
+    // 2.) Scrape page data (title & resource).
+    const updated = await getPageScrapeValues(resourceUrl, isEclkc);
+    if (updated === 0) {
+      // We want to cause a retry.
+      auditLogger.info(`Resource Queue: Warning, unable to retrieve resource title for resource '${resourceUrl}'.`);
+      return ({ status: httpCodes.NOT_FOUND, data: { url: resourceUrl } });
+    }
+
+    // 3.) Get metadata (if ECLKC resource).
+    await getMetadataValues(resourceId, resourceUrl, isEclkc);

```

Comment:
Feel free to ignore, but I think this would read better if it was like 
```js
if(isEclkc) {
  await getMetadataValues(resourceId, resourceUrl)
}
```

Reply:
Good point.